pkg/gardenlet/controller/backupentry/add.go

```diff
@@ -138,3 +139,21 @@ func getBackupBucketLastOperation(obj client.Object) *gardencorev1beta1.LastOper
 
 	return backupBucket.Status.LastOperation
 }
+
+// restoreAfterMigratePredicate is a predicate which returns false if the core.gardener.cloud/v1beta1.BackupEntry has been
+// successfully migrated and does not have the `gardener.cloud/operation: restore` annotation.
+// It returns true in all other cases.
+func restoreAfterMigratePredicate(obj client.Object) bool {
```

Comment:
```suggestion
// migrationOrRestorePending is a predicate which returns true if the core.gardener.cloud/v1beta1.BackupEntry is being migrated or restored.
func migrationOrRestorePending(obj client.Object) bool {
```

Nit

Reply:
The way that this currently works is that the predicate returns `true` when the last operation is NOT `Migrate/Succeeded` because we want to enqueue reconciliation requests in those cases. 

We only want to block reconciliation requests (return `false`) if the operation IS `Migrate/Succeeded` and the resource does NOT have the `gardener.cloud/operation=restore` annotation.

In line with your suggestion, I could rename the function to better indicate this:
```
// restoreNotPending is a predicate which returns true if the `core.gardener.cloud/v1beta1.BackupEntry` is not waiting to be restored after being migrated.
restoreNotPending(...)
```
I wondered whether to use `predicates.Not` to invert the function, instead of having `Not` be part of the name, however the https://github.com/plkokanov/gardener/blob/556ea34276a51de759e5c0574f9a15ed845c77fa/pkg/gardenlet/controller/backupentry/add.go#L106 function only takes a `func(obj client.Obj) bool` as second argument. 
wdyt?

Reply:
We discussed this with @timuthy in slack and decided to use the following:
```
// backupEntryPredicate is a predicate which returns true if the core.gardener.cloud/v1beta1.BackupEntry has not yet been successfully migrated or has the `gardener.cloud/operation: restore` annotation.
func backupEntryPredicate(obj client.Object) bool {
```